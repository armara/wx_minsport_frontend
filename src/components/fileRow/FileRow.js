import React from 'react'
import PropTypes from 'prop-types'

import Button from 'components/button/Button'

import 'components/fileRow/FileRow.scss'

const FileRow = ({ fileName, buttonText, onClick }) => (
  <>
    <span className="file-row_file-block">
      <i className="ic-pdf" />
      <span>{fileName}</span>
    </span>

    <Button className="file-row_button" onClick={onClick}>
      {buttonText}
    </Button>
  </>
)

FileRow.defaultProps = {
  onClick: () => {},
}

FileRow.propTypes = {
  onClick: PropTypes.func,
  buttonText: PropTypes.string.isRequired,
  fileName: PropTypes.string.isRequired,
}

export default FileRow
