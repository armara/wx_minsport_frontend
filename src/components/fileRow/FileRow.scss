@import "styles/mixins";

.file-row_file-block {
  display: flex;
  align-items: center;

  span {
    @include roboto-regular-text(12px, 18px, $evil-gray);
  }

  i {
    margin-right: 8px;
  }
}

.file-row_button {
  @include form-reset-button();
  @include roboto-regular-text();
}