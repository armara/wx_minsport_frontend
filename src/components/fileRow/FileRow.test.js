import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import FileRow from 'components/fileRow/FileRow'

configure({ adapter: new Adapter() })

describe('<FileRow />', () => {
  it('renders correctly default props', () => {
    const tree = renderer
      .create(
        <FileRow fileName="Peter's embassy declaration" buttonText="button" />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
