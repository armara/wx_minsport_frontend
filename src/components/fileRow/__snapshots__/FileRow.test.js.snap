// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileRow /> renders correctly default props 1`] = `
Array [
  <span
    className="file-row_file-block"
  >
    <i
      className="ic-pdf"
    />
    <span>
      Peter's embassy declaration
    </span>
  </span>,
  <button
    className="file-row_button button base-button"
    disabled={false}
    id={null}
    onClick={[Function]}
    title=""
    type="button"
  >
    
    button
  </button>,
]
`;
